testapp/tests.py

```diff
@@ -60,86 +62,187 @@ def test_create_change_request(self, mock_get_snow_group_guid, mock_pysnow):
+        }
+
+        fake_resource.update.return_value = retval
+        self.mock_pysnow_client.resource.return_value = fake_resource
         mock_pysnow.Client.return_value = self.mock_pysnow_client
 
         ret_val = self.change_request_handler.update_change_request(fake_change_order, payload='{"foo": "bar"}')
         self.assertEqual(fake_change_order.state, ChangeRequest.TICKET_STATE_COMPLETE)
-        self.assertEqual(ret_val, {'state': ChangeRequest.TICKET_STATE_COMPLETE})
+        self.assertEqual(fake_change_order.title, retval['short_description'])
+        self.assertEqual(fake_change_order.description, retval['description'])
+        self.assertEqual(fake_change_order.assignment_group_guid, retval['assignment_group']['value'])
+        self.assertEqual(ret_val, retval)
 
-    def test_update_change_request_raises_exception(self, mock_pysnow):
-        fake_query = mock.MagicMock()
+    def test_update_change_request_raises_exception_for_http_error(self, mock_pysnow):
+        fake_resource = mock.MagicMock()
         fake_change_order = mock.MagicMock()
 
-        fake_query.update.return_value = {'error': '3'}
-        self.mock_pysnow_client.query.return_value = fake_query
+        fake_exception = HTTPError()
+        fake_exception.response = mock.MagicMock()
+        fake_exception.response.text.return_value = 'Foobar'
+
+        fake_resource.update.side_effect = fake_exception
+
+        self.mock_pysnow_client.resource.return_value = fake_resource
         mock_pysnow.Client.return_value = self.mock_pysnow_client
 
         with self.assertRaises(ChangeRequestException):
             self.change_request_handler.update_change_request(fake_change_order, payload='{"foo": "bar"}')
 
-    def test__get_snow_group_guid_cached_result(self, mock_pysnow):
-        self.change_request_handler.group_guid_dict['foo'] = 'bar'
-        self.assertEqual(self.change_request_handler._get_snow_group_guid('foo'), 'bar')
+    def test_update_change_request_raises_exception_for_error_in_result(self, mock_pysnow):
+        fake_resource = mock.MagicMock()
+        fake_change_order = mock.MagicMock()
 
-    def test__get_snow_group_guid(self, mock_pysnow):
-        fake_query = mock.MagicMock()
-        fake_query.get_one.return_value = {'sys_id': 'yo'}
-        self.mock_pysnow_client.query.return_value = fake_query
+        fake_resource.update.return_value = {'error': '3'}
+        self.mock_pysnow_client.resource.return_value = fake_resource
+        mock_pysnow.Client.return_value = self.mock_pysnow_client
+
+        with self.assertRaises(ChangeRequestException):
+            self.change_request_handler.update_change_request(fake_change_order, payload='{"foo": "bar"}')
+
+    def test_get_snow_group_guid_cached_result(self, mock_pysnow):
+        self.change_request_handler.group_guid_dict['foo'] = 'bar'
+        self.assertEqual(self.change_request_handler.get_snow_group_guid('foo'), 'bar')
```

Comment:
This test has the same code smell as I mentioned above with needing to have intimate knowledge of the inner workings of the code. I think it would be better to mock the pysnow result, and ensure that pysnow is only called on the first attempt to retrieve the guid. This same technique should probably be used in other tests. In fact, this could even justify the continued use of `self.mock_pysnow_client`, since that would be functionality which would be necessary for many of the tests.